x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -1083,47 +1135,47 @@ private void doUpdateApiKey(
         if (currentDocVersion.before(targetDocVersion)) {
```

Comment:
With the bulk update function, this method can run many times in a loop. I think it is better if we check whether debug is enabled first, i.e.
```suggestion
        if (logger.isDebugEnabled() && currentDocVersion.before(targetDocVersion)) {
```
Same for the trace logging above.